nibabel/casting.py

```diff
@@ -796,3 +796,10 @@ def ulp(val=np.float64(1.0)):
         fl2 = info['minexp']
     # 'nmant' value does not include implicit first bit
     return 2 ** (fl2 - info['nmant'])
+
+
+# Ported from np.compat
+def asstr(s):
+    if isinstance(s, bytes):
+        return s.decode('latin1')
+    return str(s)
```

Comment:
... in other words, unless in each instance that `asstr` is used we know something is always either bytes or str (in which case, why use this func at all???) then the places `asstr` is used we'll need to put this conditional in. But maybe that's cleaner... I'm happy either way!